hydra-cluster/test/Test/EndToEndSpec.hs

```diff
@@ -313,11 +335,62 @@ spec = around showLogsOnFailure $ do
         failAfter 5 $ do
           version <- readCreateProcess (proc "hydra-node" ["--version"]) ""
           version `shouldSatisfy` (=~ ("[0-9]+\\.[0-9]+\\.[0-9]+(-[a-zA-Z0-9]+)?" :: String))
+      fit "hydra-node logs it's command line arguments" $ \tracer -> do
+        failAfter 60 $
+          withTempDir "temp-dir-to-check-hydra-logs" $ \dir -> do
+            withCardanoNodeDevnet (contramap FromCardanoNode tracer) dir $ \node -> do
+              let hydraSK = dir </> "hydra.sk"
+                  cardanoSK = dir </> "cardano.sk"
+                  cardanoLedgerGenesisFile = dir </> "genesis.json"
+                  cardanoLedgerProtocolParametersFile = dir </> "protocol-parameters.json"
+
+              writeFileBS cardanoLedgerGenesisFile ""
+              writeFileBS cardanoLedgerProtocolParametersFile ""
+              let ledgerCfg =
+                    CardanoLedgerConfig
+                      { cardanoLedgerGenesisFile
+                      , cardanoLedgerProtocolParametersFile
+                      }
+              hydraSKey :: SigningKey HydraKey <- generate arbitrary
+              cardanoSKey <- generate genSigningKey
+              void $ writeFileTextEnvelope hydraSK Nothing hydraSKey
+              void $ writeFileTextEnvelope cardanoSK Nothing cardanoSKey
+              hydraScriptsId <- publishHydraScriptsAs node Faucet
+              arbitraryOptions <- generate arbitrary
+              -- change the arbitrary options to suit our needs
+              let runOptions =
+                    arbitraryOptions
+                      { verbosity = Verbose ("HydraNode-" <> show (nodeId (HO.nodeId arbitraryOptions)))
+                      , host = "127.0.0.1"
+                      , HO.nodeId = HO.nodeId arbitraryOptions
+                      , apiHost = "127.0.0.1"
+                      , hydraSigningKey = hydraSK
+                      , peers = []
+                      , hydraVerificationKeys = []
+                      , HO.hydraScriptsTxId = hydraScriptsId
+                      , ledgerConfig = ledgerCfg
+                      , chainConfig =
+                          (chainConfig arbitraryOptions)
+                            { cardanoVerificationKeys = []
+                            , cardanoSigningKey = cardanoSK
+                            , HO.nodeSocket = nodeSocket node
+                            , HO.networkId = networkId node
+                            , startChainFrom = Nothing
+                            }
+                      }
+              let args = toArgs runOptions
+              -- REVIEW: running the hydra-node with supplied options fails with hydra-node:
+              --         user error (Error in $: not enough input)
+              -- so we're just grabbing the output here to determine if the logs match our provided options
+              (_, Just nodeOutput, _, _) <- createProcess (proc "hydra-node" args){std_out = CreatePipe}
```

Comment:
IMO using arbitrary options (or some default options + the non-optional ones), grabbing output and then killing it after seeing the output we expected should work without all the setup above.